### PR TITLE
Media Foundation/V4L2 grabber ...

### LINKED ISSUE #1263: [FR] Support for password managers
## Feature request

The ability to use password managers for signing in. Currently, the password input field has no "name" attribute, causing it to be not detected by password managers.

#### What problem does this feature solve?

Easier login. Especially handy since Hyperion currently doesn't keep your session. (Also suggested as FR at #858)

#### What does the proposed API look like?

N/A

#### How should this be implemented in your opinion?

The only required change would to add the following attribute to the password input element:

name="password"

#### Are you willing to work on this yourself?

I wouldn't mind creating a PR, since it's very easy to fix.

### PR DESCRIPTION
**Pull request by Paulchen-Panther**
**Summary**

- Added:
  - Windows Media Foundation USB grabber (incl. Media Foundation transform/Turbo-JPEG scaling)
  - Linux V4L2 Grabber now supports the following formats: NV12, YUV420
  - Image flipping ability in ImageResampler/Turbo-JPEG
  - UI: Simplified screens for non-expert usage, do only show elements relevant
  - Discover Grabbers available (incl. their capabilities for selection), not supported grabbers are not presented
  Note: Screen capturing on Wayland is not supported (given the Wayland security architecture)
  - USB Grabber, new ability to configure hardware controls (brightness, contrast, saturation, hue), as well as populating defaults
  - Configuration item ranges are automatically adopted based on grabber capabilities, 
  - Grabbers can only be saved with a valid configuration
  - Standalone grabbers: Added consistent options/capabilities for standalone grabbers, debug logging support
  - Screen grabbers: Allow to set capture frequency, size decimation and cropping across all grabber types
  - Screen grabber: QT-Grabber allows to capture individual displays or all displays in a multi-display setup
  - Display Signal Detection area in preview (expert users)
  - UI: Only show CEC detection, if supported by platform

- Changed:
  - The Dashboard is now a one-stop control element to control instances and link into configuration areas
  - LED Instance independent configuration objects (e.g. capturing hardware) are now separated out in the menu
  - New menu item "Sources"  per LED instances configuration  to enable/disable screen or usb grabber per instance

  - Custom effect separation in the systray menu
  - Changed configuration items are migrated to new schemes
  - Image frames from QT grabber are copied line-based

- Fixed:
  - UI Allow to have password handled by Password-Manager (Issue #1263) 
  - Active grabbers are displayed correctly after updating the WebUI
  - Python linking on MacOS (brew link/unlink)
  - Issue #1160
  - Framebuffer grabber is deactivated in case of error
  - Grabbers use now precise timings for better timing accuracy
  - Fixed effect freezing during startup

- Removed:
  - The commented JSON config

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [x] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of web configuration, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing setups:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's body (e.g. `Fixes: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [ ] Related documents have been updated (docs/docs/en)
- [ ] Related tests have been updated

**PLEASE DON'T FORGET TO ADD YOUR CHANGES TO CHANGELOG.MD**
- [ ] Yes, CHANGELOG.md is also updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

Fixes #1263, #1160, #1266, #1220, #1189,

### CODE REVIEW

**Comment by rudihorn** [comment]
I tried building this PR on unix and am having issues with the `libjpeg-turbo` dependency. The version of this library delivered by apt on Raspbian is too outdated and does not include some of the functions used, so I built it from source code. It seems to now compile but fails at linking because the cmake dependencies file only references the library when compiling from windows and not on Unix. The CI bot seems to skip this issue by compiling without `libjpeg-turbo`.

Is this PR supposed to slowly be in a compilable state already or should we wait a bit longer?

**Comment by Paulchen-Panther** [comment]
> I tried building this PR on unix and am having issues with the `libjpeg-turbo` dependency.

Specifying Unix as the build system is wide-ranging. If you tell us your hardware and the operating system you are compiling on, we could narrow down the problem.

> The version of this library delivered by apt on Raspbian is too outdated and does not include some of the functions used, so I built it from source code

Which Raspbian version is used. Apart from the fact that the current operating system for the RPi is called "Raspberry Pi OS".

> It seems to now compile but fails at linking because the cmake dependencies file only references the library when compiling from windows and not on Unix

Unfortunately, I cannot confirm this statement. 

> The CI bot seems to skip this issue by compiling without `libjpeg-turbo`.

That's right. The Docker build images do not yet contain turbo-jpeg libraries. But this is still changing

**Comment by rudihorn** [comment]
Sorry yes I was using Debian Buster on a raspberry pi 2+. I have now resolved this issue as I was accidentally specifying the `libjpeg-turbo` library directory rather than the library itself. So to build this after compiling the library to `/opt/libjpeg-turbo` directory I run the cmake command:

`cmake -DCMAKE_INSTALL_PREFIX=/home/pi/apps -DTurboJPEG_INCLUDE_DIRS=/opt/libjpeg-turbo/include -DTurboJPEG_LIBRARY=/opt/libjpeg-turbo/lib32/libturbojpeg.so ..`

**Comment by Paulchen-Panther** [comment]
@rudihorn 
I have updated the CI. You can download the current build (artifact) here: 
https://github.com/hyperion-project/hyperion.ng/actions/runs/886664639

File: include/grabber/FramebufferWrapper.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 ///
 /// The FramebufferWrapper uses an instance of the FramebufferFrameGrabber to obtain ImageRgb's from the
-/// displayed content. This ImageRgb is processed to a ColorRgb for each led and commmited to the
+/// displayed content. This ImageRgb is processed to a ColorRgb for each led and committed to the
 /// attached Hyperion.
```
Comment (tpmodding) [inline include/grabber/FramebufferWrapper.h:9]: steinig mich bitte nicht, kann das nicht mit in Zeile 8? :D

File: libsrc/grabber/dispmanx/DispmanxFrameGrabber.cpp [hunk 2]
```diff
@@ -64,152 +72,219 @@ void DispmanxFrameGrabber::freeResources()
...
+			VC_IMAGE_RGBA32,
+			width,
+			height,
+			&vc_nativeImageHandle);
 		assert(_vc_resource);
 
-		// Define the capture rectangle with the same size
-		vc_dispmanx_rect_set(&_rectangle, 0, 0, width, height);
-		return true;
+		if (_vc_resource != 0)
```
[the remark this answers is not in the document]
Comment (Lord-Grey) [reply libsrc/grabber/dispmanx/DispmanxFrameGrabber.cpp:91]: Ja. Lies mal den Code.
1. Falls es noch eine allokierte gibt, dann lösche sie
2. Jetzt wird  wieder eine resource angelegt, _vc_resource = vc_dispmanx_resource_create
Und wenn das erfolgreich war  (d.h. _vc_resource  != 0), dann setze das capture rect

File: libsrc/grabber/amlogic/AmlogicWrapper.cpp [hunk 3]
```diff
@@ -1,9 +1,11 @@
 #include <grabber/AmlogicWrapper.h>
 
-AmlogicWrapper::AmlogicWrapper(unsigned grabWidth, unsigned grabHeight)
-	: GrabberWrapper("AmLogic", &_grabber, grabWidth, grabHeight)
-	, _grabber(grabWidth, grabHeight)
-{}
+AmlogicWrapper::AmlogicWrapper(int pixelDecimation,	int updateRate_Hz)
+	: GrabberWrapper("AmLogic", &_grabber, updateRate_Hz)
```
[the remark this answers is not in the document]
Comment (Lord-Grey) [reply libsrc/grabber/amlogic/AmlogicWrapper.cpp]: Done

File: include/grabber/X11Wrapper.h [hunk 4]
```diff
@@ -10,24 +10,26 @@
 
 ///
 /// The X11Wrapper uses an instance of the X11Grabber to obtain ImageRgb's from the
-/// displayed content. This ImageRgb is processed to a ColorRgb for each led and commmited to the
+/// displayed content. This ImageRgb is processed to a ColorRgb for each led and committed to the
 /// attached Hyperion.
```
[the remark this answers is not in the document]
Comment (Lord-Grey) [reply include/grabber/X11Wrapper.h]: Done

File: include/grabber/OsxWrapper.h [hunk 5]
```diff
@@ -5,7 +5,7 @@
 
 ///
 /// The OsxWrapper uses an instance of the OsxFrameGrabber to obtain ImageRgb's from the
-/// displayed content. This ImageRgb is processed to a ColorRgb for each led and commmited to the
+/// displayed content. This ImageRgb is processed to a ColorRgb for each led and committed to the
 /// attached Hyperion.
```
[the remark this answers is not in the document]
Comment (Lord-Grey) [reply include/grabber/OsxWrapper.h]: Done

File: include/grabber/FramebufferWrapper.h (hunk 1 above)
Comment (Lord-Grey) [reply include/grabber/FramebufferWrapper.h:9]: Done

File: libsrc/grabber/dispmanx/DispmanxFrameGrabber.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (tpmodding) [reply libsrc/grabber/dispmanx/DispmanxFrameGrabber.cpp:91]: Ah danke!!

**Pull request merged**